### PR TITLE
Issue 3546: Add "overridable" to the configuration variable description.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
This is for #3546.

This is a bit more complex than it seems it should, because the information needed was only in a `static` array in "InkAPI.cc". This PR moves that array out to a separate file ("txn_args.cc") which is
then `#include`d into the original location in "InkAPI.cc" and into "config.cc" in `traffic_ctl`.

To mark the file as shared between two commands, it was put in to a new "shared" directory. I think "tscore" is the wrong place because that would create circular dependencies, as this array depends
on types defined over in "proxy/http". I also considered whether to `#include` or not, but putting it in
a separate library and having to also have a header file seemed a bit too much overhead. If other files end up this way, that should be rethought.

### CODE REVIEW

File: src/traffic_ctl/config.cc [hunk 1]
```diff
@@ -51,6 +55,8 @@ struct CtrlMgmtRecordDescriptionList : CtrlMgmtList<RecordDescriptionPolicy> {
   }
 };
 
+#include "../shared/txn_vars.cc"
```
Comment (zwoop) [inline src/traffic_ctl/config.cc]: This is slightly wonky, do we really need / want to include .cc files ? Seeing this, maybe src/shared should compile to a .a (not .so), and we add a include/shared for the corresponding prototypes?

Comment (SolidWallOfCode) [reply src/traffic_ctl/config.cc]: Hahaha, the bike shedder arrives!

I'm not sure. Is it worth the set up for a single file?

In any case, I don't like the ".a" approach. If you want a change, I would create a header file, make the `Overridable_Map` not static, and grab the file via "Makefile.inc" reaching over to "../shared/txn-vars.cc".

What would the directory structure be?
-shared
  -src
    -txn_vars.cc
  -include
    -txn_vars.h

or
-shared
  -txn_vars.cc
  -include
    -txn_vars.h

**Comment by SolidWallOfCode** [comment]
I updated the PR with documentation updates, both for this PR and some other cleanup.

**Comment by SolidWallOfCode** [comment]
[approve ci clang-analyzer]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**